### PR TITLE
Migrate node_io to null safety

### PR DESCRIPTION
**Pull request by jathak**
This migrates the node_io code itself, but not the tests

### CODE REVIEW

File: node_io/lib/src/http_server.dart [hunk 1]
```diff
@@ -415,11 +416,14 @@ class NodeHttpRequest implements io.HttpRequest, HasReadable {
...
+    List<int> Function()? orElse,
   }) {
-    return _delegate.firstWhere(test, orElse: () {
-      return Uint8List.fromList(orElse());
-    });
+    return _delegate.firstWhere(test,
+        orElse: orElse == null
+            ? null
+            : () {
+                return Uint8List.fromList(orElse());
```
Comment (Awjin) [inline node_io/lib/src/http_server.dart]: Nit here and throughout this file: This could be an arrow function

File: node_io/lib/node_io.dart [hunk 2]
```diff
@@ -72,11 +72,9 @@ set exitCode(int value) {
 ///
 /// On Windows the exit code can be set to any 32-bit value. However some of
 /// these values are reserved for reporting system errors like crashes.
-void exit([int code]) {
-  if (code is! int) {
-    throw ArgumentError('Integer value for exit code expected');
-  }
+Never exit(int code) {
```
Comment (Awjin) [inline node_io/lib/node_io.dart:75]: Is there a reason for this behavior change?

Comment (jathak) [reply node_io/lib/node_io.dart:75]: I'm trying to match the API for `dart:io`, which has `exit` take in a non-nullable integer and return the new `Never` type.

File: node_io/lib/src/http_server.dart (hunk 1 above)
Comment (jathak) [reply node_io/lib/src/http_server.dart]: Done

Comment (jathak) [reply node_io/lib/src/http_server.dart]: Done

File: node_io/lib/node_io.dart (hunk 2 above)
Comment (Awjin) [reply node_io/lib/node_io.dart:75]: I see. Worth documenting in the changelog?

**Review by Awjin — APPROVED** [review]

File: node_io/lib/node_io.dart (hunk 2 above)
Comment (jathak) [reply node_io/lib/node_io.dart:75]: Done

**Pull request merged**